packages/plugins/experimental/plugin-map/src/components/MapDetectLocations.ts

```diff
@@ -85,7 +85,10 @@ export const useMapDetectLocations = (map: MapType): Marker[] => {
 export default useMapDetectLocations;
 
 const hasId = (obj: unknown): obj is { id: string } =>
-  typeof obj === 'object' && obj !== null && 'id' in obj && typeof obj.id === 'string';
```

Comment:
what the scenario where it fails `'id' in obj` but passes `Object.hasOwnProperty(obj, 'id')`?

Reply:
Great question. Strange stuff is afoot:
![image](https://github.com/user-attachments/assets/2a23f525-bbba-4cf8-bb62-1bfc9efd607b)

I'll investigate it more monday.

Reply:
Now that `hasId` has been removed in https://github.com/dxos/dxos/pull/7681 we don't need to look into this more.